### PR TITLE
chore: use actions variables to pass Cardano Services urls

### PR DESCRIPTION
**Pull request by pczeglik-iohk**
# Checklist

- [x] JIRA - [LW-10078](https://input-output.atlassian.net/browse/LW-10078)
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

Use Cardano services urls for Mainnet, Preprod, and Preview from Github actions variables instead of hardcoded values in .env.defaults file

image

Default Cardano Services values for CI jobs are **DEV** `CARDANO_SERVICES_URL_DEV_*`, whereas for the release branches we use **LIVE** `CARDANO_SERVICES_URL_LIVE_*`

[LW-10078]: https://input-output.atlassian.net/browse/LW-10078?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/browser-extension-wallet/webpack.app.dev.js [hunk 1]
```diff
@@ -5,7 +5,13 @@ const appConfig = require('./webpack.common.app');
 require('dotenv-defaults').config({
   path: './.env',
   encoding: 'utf8',
-  defaults: process.env.BUILD_DEV_PREVIEW === 'true' ? './.env.developerpreview' : './.env.defaults'
+  defaults: process.env.BUILD_DEV_PREVIEW === 'true' ? './.env.developerpreview' : './.env.defaults',
+  override: true
```
Comment (pczeglik-iohk) [inline apps/browser-extension-wallet/webpack.app.dev.js:9]: https://www.npmjs.com/package/dotenv#override

File: apps/browser-extension-wallet/webpack.app.dev.js [hunk 2]
```diff
@@ -5,7 +5,13 @@ const appConfig = require('./webpack.common.app');
...
 });
 
+console.log('CARDANO SERVICES URL MAINNET: ', process.env.CARDANO_SERVICES_URL_MAINNET);
```
Comment (pczeglik-iohk) [inline apps/browser-extension-wallet/webpack.app.dev.js:12]: https://github.com/input-output-hk/lace/actions/runs/8520340496/job/23336261120#step:4:1211

```
@lace/browser-extension-wallet
 | CARDANO SERVICES URL MAINNET:  https://dev-mainnet.lw.iog.io/
 | CARDANO SERVICES URL PREPROD:  https://dev-preprod.lw.iog.io/
 | CARDANO SERVICES URL PREVIEW:  https://dev-preview.lw.iog.io/
 | CARDANO SERVICES URL SANCHONET:  https://backend.dev-sanchonet.eks.lw.iog.io/
```

Comment (pczeglik-iohk) [reply apps/browser-extension-wallet/webpack.app.dev.js:12]: this is ☝️ for prod build in CI

File: apps/browser-extension-wallet/webpack.app.prod.js [hunk 3]
```diff
@@ -5,7 +5,13 @@ const appConfig = require('./webpack.common.app');
 require('dotenv-defaults').config({
   path: './.env',
   encoding: 'utf8',
-  defaults: process.env.BUILD_DEV_PREVIEW === 'true' ? './.env.developerpreview' : './.env.defaults'
+  defaults: process.env.BUILD_DEV_PREVIEW === 'true' ? './.env.developerpreview' : './.env.defaults',
+  override: true
 });
 
+console.log('CARDANO SERVICES URL MAINNET: ', process.env.CARDANO_SERVICES_URL_MAINNET);
```
Comment (pczeglik-iohk) [inline apps/browser-extension-wallet/webpack.app.prod.js:12]:
```
@lace/browser-extension-wallet
 | CARDANO SERVICES URL MAINNET:  https://dev-mainnet.lw.iog.io
 | CARDANO SERVICES URL PREPROD:  https://dev-preprod.lw.iog.io
 | CARDANO SERVICES URL PREVIEW:  https://dev-preview.lw.iog.io
 | CARDANO SERVICES URL SANCHONET:  https://backend.dev-sanchonet.eks.lw.iog.io
 | CARDANO SERVICES URL MAINNET:  https://dev-mainnet.lw.iog.io
 | CARDANO SERVICES URL PREPROD:  https://dev-preprod.lw.iog.io
 | CARDANO SERVICES URL PREVIEW:  https://dev-preview.lw.iog.io
 | CARDANO SERVICES URL SANCHONET:  https://backend.dev-sanchonet.eks.lw.iog.io
```

this is ☝️ for prod build in CI yarn build

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -13,32 +13,41 @@ jobs:
...
       - name: Decrypt test data
         working-directory: ./packages/e2e-tests
         run: ./decrypt_secret.sh
         env:
           WALLET_1_PASSWORD: ${{ secrets.WALLET_PASSWORD_TESTNET }}
+
       - name: Build dist version of Lace
         uses: ./.github/shared/build
         with:
+          CARDANO_SERVICES_URL_MAINNET: ${{ startsWith(github.ref, 'refs/heads/release') && vars.CARDANO_SERVICES_URL_LIVE_MAINNET || vars.CARDANO_SERVICES_URL_DEV_MAINNET }}
```
Comment (pczeglik-iohk) [inline .github/workflows/ci.yml:26]: For all CI jobs / actions we use **DEV** urls, but for `release` branches we use **LIVE**

**Review by rhyslbw — COMMENTED** [review]
There's lots of repetition in these workflows, can we lift it to an action?

**Pull request closed without merging**